### ISSUE TITLE
Ability to add more than one 'other' value

### ISSUE THREAD

**Issue by twelch** [Ecotrust/solomon#65]
For shellfish resource types and other questions, there is a strong chance of needing to be able to add more than one 'other', especially since the paper form leaves a number of blanks spots to write in.

This could be done as an 'add another', etc.

**Comment by eknuth** [comment]
Ugh, I imagine we need this on grid selects too.  I'm not even quite sure how to do the ui for the first "other" option.  Did we talk about a modal or something for that?

**Comment by twelch** [comment]
Yeah we discussed a modal.  You can wait on this task for the moment
